OpenAPI/kiota/tutorials/dotnet-dependencyinjection.md

```diff
@@ -0,0 +1,288 @@
+# dotnet add package Microsoft.Kiota.Serialization.Form
+# dotnet add package Microsoft.Kiota.Serialization.Text
+# dotnet add package Microsoft.Kiota.Serialization.Multipart
+```
+
+## Generate the API client
+
+Kiota generates API clients from OpenAPI documents. Create a file named _github-releases.yml_ and add the following.
+
+```yaml
+openapi: 3.0.3
+info:
+  title: GitHub Release API
+  version: 1.0.0
+servers:
+  - url: https://api.github.com
+    description: GitHub API
+    variables:
+      version:
+        default: '2022-11-28'
+        description: GitHub API version
+paths:
+  /repos/{owner}/{repo}/releases/latest:
+    get:
+      summary: Get the latest release
+      parameters:
+        - name: owner
+          in: path
+          required: true
+          description: The owner of the repository
+          schema:
+            type: string
+        - name: repo
+          in: path
+          required: true
+          description: The repository name
+          schema:
+            type: string
+      responses:
+        200:
+          description: Success!
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/github.release"
+        404:
+          description: Not found
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/github.error"
+
+components:
+  schemas:
+    github.release:
+      type: object
+      properties:
+        id:
+          type: integer
+          description: The release ID
+        tag_name:
+          type: string
+          description: The release tag name
+        name:
+          type: string
+          description: The release name
+        body:
+          type: string
+          description: The release body
+        created_at:
+          type: string
+          format: date-time
+          description: The release creation date
+        published_at:
+          type: string
+          format: date-time
+          description: The release publication date
+    github.error:
+      type: object
+      properties:
+        message:
+          type: string
+          description: The error message
+        documentation_url:
+          type: string
+          description: The error documentation URL
+```
+
+You can then use the [Kiota](/openapi/kiota/install?tabs=bash&wt.mc_id=SEC-MVP-5004985#install-as-net-tool) command line tool to generate the API client classes. We are using the [deserializer](/openapi/kiota/using#--deserializer---ds) and [serializer](/openapi/kiota/using#--serializer---s) options to specify the serialization format, to only use JSON serialization.
+
+```bash
+kiota generate -l csharp -d github-releases.yml -c GitHubClient -n GitHub.ApiClient -o ./GitHub --deserializer Microsoft.Kiota.Serialization.Json.JsonParseNodeFactory --serializer Microsoft.Kiota.Serialization.Json.JsonSerializationWriterFactory
```

Comment:
I'd leave the defaults for simplicity and remove the serializer and deserializer arguments. Same for the dependencies earlier.
Other documentation pages explore/will explore the serialization mechanisms and the dependencies, and the choices they can make.